haec-user-service/src/modules/user/services/user.service.ts

```diff
@@ -46,14 +46,51 @@ export class UserService {
           email: input.email,
         },
       });
-
       if (await bcrypt.compare(input.password, user.hashedPass)) {
         return true;
       }
     } catch (error) {
       console.error(error);
     }
-
     return false;
   }
+
+// This will log the user ** IN **
+
+  public async loginUser(input: {
+    email: string;
+    password: string;
+  }): Promise<string> {
+    const verified = await this.verifyUser({ email: input.email, password: input.password})
+    if (verified == true){
+      // Generates random 21 character token with Crypto
+      const Crypto = require('crypto')
+      function randomString(size = 21) {
+        return Crypto
+          .randomBytes(size)
+          .toString('base64')
+          .slice(0, size)
+      }
+      const token = randomString()
+      await this.prisma.user.update({
+        where: { email: input.email },
+        data: { token: token } })
+      return token
+  }
+  else {
+    console.log("error")
+    throw new Error()
```

Comment:
Couple things here:

- With error messages, I'd be specific as to what the error is (will make your life SO much easier in the long run). I'd change this to something like `Error logging in user`.
- Second, you can actually put the same string into the `new Error()` just like you do with `console.log`! Can do something like:

```ts
console.log('this is some error text')
throw new Error('this is some error text')
```

Reply:
That makes a ton of sense. If I'm going to have it throw an error, it should be descriptive.